packages/advanced-logic/src/extensions/payment-network/any-to-eth-proxy.ts

```diff
@@ -0,0 +1,119 @@
+import { ExtensionTypes, RequestLogicTypes } from '@requestnetwork/types';
+import EthereumFeeProxyPaymentNetwork from './ethereum/fee-proxy-contract';
+import { supportedCurrencies } from './chainlink-supported-currencies';
+
+const CURRENT_VERSION = '0.1.0';
+// Default network if the storage data does not give any
+const DEFAULT_NETWORK = 'mainnet';
+
+export default class AnyToEthProxyPaymentNetwork extends EthereumFeeProxyPaymentNetwork {
+  public constructor(
+    extensionId: ExtensionTypes.ID = ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ETH_PROXY,
+    currentVersion: string = CURRENT_VERSION,
+  ) {
+    super(extensionId, currentVersion, Object.keys(supportedCurrencies));
+  }
+
+  /**
+   * Creates the extensionsData to create the extension ETH fee proxy contract payment detection
+   *
+   * @param creationParameters extensions parameters to create
+   *
+   * @returns IExtensionCreationAction the extensionsData to be stored in the request
+   */
+  public createCreationAction(
+    creationParameters: ExtensionTypes.PnAnyToEth.ICreationParameters,
+  ): ExtensionTypes.IAction {
+    const network = creationParameters.network;
+    if (!network) {
+      throw Error('network is required');
+    }
+    if (!supportedCurrencies[network]) {
+      throw Error(`network ${network} not supported`);
+    }
+    return super.createCreationAction(creationParameters);
+  }
+
+  /**
+   * Applies a creation extension action
+   *
+   * @param extensionAction action to apply
+   * @param timestamp action timestamp
+   *
+   * @returns state of the extension created
+   */
+  protected applyCreation(
+    extensionAction: ExtensionTypes.IAction,
+    timestamp: number,
+  ): ExtensionTypes.IState {
+    if (!extensionAction.parameters.network || extensionAction.parameters.network.length === 0) {
+      throw Error('network is required');
+    }
+
+    const feePNCreationAction = super.applyCreation(extensionAction, timestamp);
+
+    return {
+      ...feePNCreationAction,
+      events: [
+        {
+          name: 'create',
+          parameters: {
+            feeAddress: extensionAction.parameters.feeAddress,
+            feeAmount: extensionAction.parameters.feeAmount,
+            paymentAddress: extensionAction.parameters.paymentAddress,
+            refundAddress: extensionAction.parameters.refundAddress,
+            salt: extensionAction.parameters.salt,
+            network: extensionAction.parameters.network || DEFAULT_NETWORK,
+            maxRateTimespan: extensionAction.parameters.maxRateTimespan,
```

Comment:
```suggestion
   ...extensionAction.parameters,
```
Not sure but it looks very straight-forward

Reply:
for this and your previous comment, we should have a validation schema instead. 

the problem with `...extensionAction.parameters` is that maybe there are other properties.

Reply:
I add it to my todo list to add a "validation schema" (can be done with the "currency PR" I will do)